openss7-0.9.2.D.SLES9_K2.6.5_rc1(1)/openss7-0.9.2.D.SLES9_K2.6.5_rc1/LiS/man/man8/strmakenodes.8.man
'\" rtp
.\" -*- nroff -*- vim: ft=nroff nocin nosi
.\"
.\" @(#) strmakenodes.8.man,v 1.1.4.4 2006/04/05 10:05:01 brian Exp
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2001-2006  OpenSS7 Corporation <http://www.openss7.com/>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to make and distribute verbatim copies of this
.\" manual provided the copyright notice and this permission notice are
.\" preserved on all copies.
.\"
.\" Permission is granted to copy and distribute modified versions of this
.\" manual under the conditions for verbatim copying, provided that the
.\" entire resulting derived work is distributed under the terms of a
.\" permission notice identical to this one
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not
.\" have taken the same level of care in the production of this manual,
.\" which is licensed free of charge, as they might when working
.\" professionally.  The author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by
.\" the source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified 2006/04/05 10:05:01 by brian
.\"
.\" -------------------------------------------------------------------------
.\"
.\" strmakenodes.8.man,v
.\" Revision 1.1.4.4  2006/04/05 10:05:01  brian
.\" - get references in right place
.\"
.\" Revision 1.1.4.3  2006/04/05 09:42:41  brian
.\" - moved .so line and updated headers
.\"
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database lis.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
.R2
.so lis.macros
.\"
.\"
.TH STRMAKENODES 8 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System Admin."
.SH NAME
.B strmakenodes
\- make primary \fISTREAMS\fP device nodes
.\"
.\"
.SH SYNOPSIS
.PP
.HP 12
\fBstrmakenodes\fR [\fIoptions\fR]
.HP 12
\fBstrmakenodes\fR {\fB-h\fR|\fB--help\fR}
.HP 12
\fBstrmakenodes\fR {\fB-V\fR|\fB--version\fR}
.HP 12
\fBstrmakenodes\fR {\fB-C\fR|\fB--copying\fR}
.PD
.\"
.\"
.SH OPTIONS
none
.\"
.\"
.SH DESCRIPTION
This manpage is under construction.
.\"
.\"
.SH DIAGNOSTICS
none
.\"
.\"
.SH BUGS
.PP
.B strmakenodes
has no known bugs.
.\"
.\"
.SH EXAMPLES
.PP
None.
.\"
.\"
.SH "SEE ALSO"
.PP
.BR STREAMS (4).
.\"
.\"
.SH COMPATIBILITY
.PP
.BR strmakenodes
is compatible with
.B LiS\c
.
.[
LiS
.]
.\"
.\"
.SH CONFORMANCE
.PP
.BR strmakenodes
does not conform to any standard.
.\"
.\"
.SH VERSIONS
.PP
This manpage was written for
.B LiS\c
.
.[
LiS
.]
.[
$LIST$
.]
.TI


openss7-0.9.2.D.SLES9_K2.6.5_rc1(1)/openss7-0.9.2.D.SLES9_K2.6.5_rc1/LiS/util/linux/strmakenodes.c
/******************************************************************* vim: ft=c

 Copyright (c) 2001-2006  OpenSS7 Corporation <http://www.openss7.com>
 Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>

 All Rights Reserved.

 This program is free software; you can redistribute it and/or modify it under
 the terms of the GNU General Public License as published by the Free Software
 Foundation; version 2 of the License.

 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
 details.

 You should have received a copy of the GNU General Public License along with
 this program; if not, write to the Free Software Foundation, Inc., 675 Mass
 Ave, Cambridge, MA 02139, USA.

 -----------------------------------------------------------------------------

 U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
 behalf of the U.S. Government ("Government"), the following provisions apply
 to you.  If the Software is supplied by the Department of Defense ("DoD"), it
 is classified as "Commercial Computer Software" under paragraph 252.227-7014
 of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
 successor regulations) and the Government is acquiring only the license rights
 granted herein (the license rights customarily provided to non-Government
 users).  If the Software is supplied to any unit or agency of the Government
 other than DoD, it is classified as "Restricted Computer Software" and the
 Government's rights in the Software are defined in paragraph 52.227-19 of the
 Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
 the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
 (or any successor regulations).

 -----------------------------------------------------------------------------

 Commercial licensing and support of this software is available from OpenSS7
 Corporation at a fee.  See http://www.openss7.com/

 *****************************************************************************/

/*
 * GENERATED BY strconf-sh (Tue Jul 11 07:06:28 MDT 2006) FROM Config.master
 * EDITS TO THIS FILE WILL BE LOST: EDIT strconf-sh INSTEAD.
 */

static char const ident[] = "$RCSfile$ $Name$($Revision$) $Date$";

#if defined(LINUX)
#	include <sys/types.h>
#	include <sys/stat.h>
#	include <sys/sysmacros.h>
#	define makedevice(maj,min) makedev(maj,min)
#elif defined(QNX)
#	include <sys/types.h>
#	include <sys/stat.h>
#	define makedevice(maj,min) makedev(1,maj,min)
#elif defined(USER)
#	include <sys/stropts.h>
#	include <sys/LiS/usrio.h>
#else
#	include <sys/types.h>
#	include <sys/stat.h>
#	define makedevice(maj,min) makedev(maj,min)
#endif
#ifdef _GNU_SOURCE
#       include <getopt.h>
#endif

#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

#if !defined(LINUX)
#	include <fcntl.h>
#endif

#if !defined(USER)
void copying(int argc, char *argv[])
{
	fprintf(stdout, "\n\
%1$s %2$s:\n\
\n\
Copyright (c) 2001-2006  OpenSS7 Corporation <http://www.openss7.com/>\n\
Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>\n\
Copyright (c) 1997       David Grothe, Gcom, Inc <dave@gcom.com>\n\
\n\
All Rights Reserved.\n\
\n\
This program is free software;  you can  redistribute  it and/or modify it under\n\
the terms of the GNU General  Public License as  published by the  Free Software\n\
Foundation; version 2 of  the  License.\n\
\n\
This program is distributed in the hope that it will be  useful, but WITHOUT ANY\n\
WARRANTY;  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A\n\
PARTICULAR PURPOSE.  See the GNU General Public License for more details.\n\
\n\
You should  have received  a copy of the GNU  General  Public License along with\n\
this program; if not, write to the Free Software Foundation, Inc., 675 Mass Ave,\n\
Cambridge, MA 02139, USA.\n\
\n\
U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on behalf\n\
of the  U.S. Government  (\"Government\"),  the following provisions apply to you.\n\
If the Software is  supplied by the Department of Defense (\"DoD\"), it is classi-\n\
fied as  \"Commercial Computer Software\"  under paragraph 252.227-7014 of the DoD\n\
Supplement  to the  Federal Acquisition Regulations  (\"DFARS\") (or any successor\n\
regulations) and the  Government  is acquiring  only the license rights  granted\n\
herein (the license  rights customarily  provided to non-Government  users).  If\n\
the Software is supplied to any unit or agency of the Government other than DoD,\n\
it is classified as  \"Restricted Computer Software\" and the  Government's rights\n\
in the  Software are defined in  paragraph 52.227-19 of the Federal  Acquisition\n\
Regulations  (\"FAR\") (or any successor  regulations) or, in the  cases of NASA, in\n\
paragraph  18.52.227-86 of the  NASA Supplement  to the  FAR (or  any  successor\n\
regulations).\n\
\n\
", argv[0], ident);
}

void version(int argc, char *argv[])
{
	fprintf(stdout, "\
\n\
%1$s %2$s:\n\
    Copyright (c) 2003-2006  OpenSS7 Corporation.  All Rights Reserved.\n\
    Copyright (c) 1997       David Grothe, Gcom, Inc <dave@gcom.com>\n\
\n\
    Distributed by OpenSS7 Corporation under GPL Version 2,\n\
    included here by reference.\n\
\n\
    See `%1$s --copying' for copying permissions.\n\
\n\
", argv[0], ident);
}

void usage(int argc, char *argv[])
{
	fprintf(stderr, "\
Usage:\n\
    %1$s [options]\n\
    %1$s {-h,--help}\n\
    %1$s {-V,--version}\n\
    %1$s {-C,--copying}\n\
", argv[0]);
}

void help(int argc, char *argv[])
{
	fprintf(stdout, "\
\n\
Usage:\n\
    %1$s [options]\n\
    %1$s {-h,--help}\n\
    %1$s {-V,--version}\n\
    %1$s {-C,--copying}\n\
Options:\n\
    -r, --remove\n\
        remove devices before making them\n\
    -h, --help, -?\n\
        print this usage information and exit\n\
    -V, --version\n\
        print version and exit\n\
    -C, --copying\n\
        print copying permission and exit\n\
\n\
", argv[0]);
}
#endif				/* !defined(USER) */

int main(int argc, char *argv[])
{
	int rslt;
	int rmopt = 0;
	char *strerror(int);

        (void) rslt;
#if !defined(USER)
	(void) umask(0);
	if (argc == 2 && argv[1][0] == '-' && argv[1][1] == 'r')
		rmopt = 1;

	while (1) {
		int c;
#ifdef _GNU_SOURCE
		int option_index = 0;
		/* *INDENT-OFF* */
		static struct option long_options[] = {
			{ "help",    no_argument, NULL, 'h'},
			{ "version", no_argument, NULL, 'V'},
			{ "copying", no_argument, NULL, 'C'},
			{ "remove",  no_argument, NULL, 'r'},
			{ "?",       no_argument, NULL, 'h'},
			{ 0, }
		};
		/* *INDENT-ON* */
		c = getopt_long_only(argc, argv, "rhVC?", long_options, &option_index);
#else				/* _GNU_SOURCE */
		c = getopt(argc, argv, "rhVC?");
#endif				/* _GNU_SOURCE */
		if (c == -1)
			break;
		switch (c) {
		case 'r':	/* -r, --remove */
			rmopt = 1;
                        break;
		case 'h':	/* -h, --help */
			help(argc, argv);
			exit(0);
		case 'V':	/* -V, --version */
			version(argc, argv);
			exit(0);
		case 'C':	/* -C, --copying */
			copying(argc, argv);
			exit(0);
		case '?':
			optind--;
		      bad_nonoption:
			if (optind < argc) {
				fprintf(stderr, "%s: illegal syntax -- ", argv[0]);
				for (; optind < argc; optind++)
					fprintf(stderr, "%s ", argv[optind]);
				fprintf(stderr, "\n");
			}
			usage(argc, argv);
			exit(2);
		}
	}
	if (optind < argc)
		goto bad_nonoption;

#endif
	(void) ident;

/* -------------------------------------------------- */

#if !defined(USER)
	(void)unlink("/dev/mux_clone");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/mux_clone", 020666, makedevice(230,233));
		if (rslt < 0)
			printf("/dev/mux_clone: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/minimux.1");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/minimux.1", 020666, makedevice(233,1));
		if (rslt < 0)
			printf("/dev/minimux.1: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/minimux.2");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/minimux.2", 020666, makedevice(233,2));
		if (rslt < 0)
			printf("/dev/minimux.2: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/clone_drvr");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/clone_drvr", 020666, makedevice(230,0));
		if (rslt < 0)
			printf("/dev/clone_drvr: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/printk");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/printk", 020666, makedevice(234,0));
		if (rslt < 0)
			printf("/dev/printk: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/sad");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/sad", 020666, makedevice(230,235));
		if (rslt < 0)
			printf("/dev/sad: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/fifo");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/fifo", 020666, makedevice(230,231));
		if (rslt < 0)
			printf("/dev/fifo: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/fifo.0");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/fifo.0", 020666, makedevice(231,0));
		if (rslt < 0)
			printf("/dev/fifo.0: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/nuls");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/nuls", 020666, makedevice(230,237));
		if (rslt < 0)
			printf("/dev/nuls: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/echo");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/echo", 020666, makedevice(230,238));
		if (rslt < 0)
			printf("/dev/echo: %s\n", strerror(-rslt));
	}
#if !defined(USER)
	(void)unlink("/dev/mux");
#endif
	if (!rmopt) {
		rslt = mknod("/dev/mux", 020666, makedevice(230,239));
		if (rslt < 0)
			printf("/dev/mux: %s\n", strerror(-rslt));
	}

/* -------------------------------------------------- */

#if !defined(USER)
	exit(0);
#else
	return (0);
#endif
}

/* 
 * vim: ft=c
 */
